### ISSUE TITLE
Expanded income and aftertax income should be options in distribution tables

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#1374]
![image](https://cloud.githubusercontent.com/assets/8301092/26267034/114e5d62-3cb6-11e7-8486-e4e11e6460f6.png)

Users may want to see average expanded income by expanded income bin or decile in TaxBrain diagnostic tables. Same with aftertax income.

**Comment by martinholmer** [comment]
I've looked into Tax-Calculator issue #1374, which says:
> Users may want to see average expanded income by expanded income bin or decile in TaxBrain diagnostic tables. Same with aftertax income.

The construction in dropq code of a TaxBrain "diagnostic" table (which is called in Tax-Calculator a "distribution" table) is fairly involved.  The data structures that are created by the dropq code have been carefully designed to meet the needs of TaxBrain developers, which is exactly as it should be.

My conclusion is that the TaxBrain developers need to start out this enhancement process by specifying exactly how they want the thirteen objects returned from the `dropq.run_nth_year_tax_calc_model` function to be changed in order to implement the #1374 enhancement.  Then, with this change specification in hand, Tax-Calculator developers can revise taxcalc code to supply revised versions of those thirteen objects.

Does this approach make sense?

@MattHJensen @feenberg @Amy-Xu @GoFroggyRun @PeterDSteinberg @brittainhard

**Comment by feenberg** [comment]
On Wed, 28 Jun 2017, Martin Holmer wrote:

> 
> I've looked into Tax-Calculator issue #1374, which says:
>
>       Users may want to see average expanded income by expanded income
>       bin or decile in TaxBrain diagnostic tables. Same with aftertax
>       income.
> 
> The construction in dropq code of a TaxBrain "diagnostic" table (which is
> called in Tax-Calculator a "distribution" table) is fairly involved. The
> data structures that are created by the dropq code have been carefully
> designed to meet the needs of TaxBrain developers, which is exactly as it
> should be.

The important thing is that we drop 3 meaningful records from each row 
returned to the user. So if we tab by expanded income we drop 3 records 
from each expanded income class, not each AGI class. Otherwise I think the 
logic should be exactly the same.

dan

**Comment by martinholmer** [comment]
Dan @feenberg said with respect to issue #1374:
 > The important thing is that we drop 3 meaningful records from each row returned to the user. So if we tab[ulate] by expanded income we drop 3 records from each expanded income class, not each AGI class.

As far as I can tell, TaxBrain never shows distributional tables with AGI bins; it shows only distributional tables with expanded-income bins.  And the dropq code does remove three records from each expanded-income bin.  So, I think your concerns are being met.

@MattHJensen @PeterDSteinberg @brittainhard

**Comment by travelmail26** [comment]
Im having issues with average tax cut size per income on the expanded EITC. When i run what i think is a close appoximation to khanna's plan, the total cost seems right, but the average tax credit is less than 1500. it should be much higher

https://www.ospc.org/taxbrain/18094/

or maybe im missing something about the averages? thanks

**Comment by martinholmer** [comment]
I've moved a question about the Brown-Khanna GAIN Act that was submitted in this issue to its own issue #1581.

**Comment by martinholmer** [comment]
@MattHJensen said in issue #1374:
> Users may want to see average expanded income by ... bin or decile in TaxBrain diagnostic tables. Same with aftertax income.

The current TaxBrain version (1.0.3) allows users to build a "Distribution Table" and a "Difference Table".

So, is your request to add to the Distribution Table two new columns: average expanded income and average after-tax expanded income?

Does it make sense to add these two new columns to the Difference Table as well?
I'm asking this because the 10/17 *New York Times* graphs are difference tables as seen in this example:


![screen shot 2017-10-19 at 12 38 48 pm](https://user-images.githubusercontent.com/12170745/31783006-c2a8db14-b4ca-11e7-8f78-97b0df31605f.png)

**Comment by MattHJensen** [comment]
@martinholmer asked two questions:

> So, is your request to add to the Distribution Table two new columns: average expanded income and average after-tax expanded income?

> Does it make sense to add these two new columns to the Difference Table as well?

Yes to both.

**Comment by martinholmer** [comment]
Is the labeling of the income groups in this *New York Times* (10/17/2017) graph slightly incorrect?

![screen shot 2017-10-19 at 12 38 48 pm](https://user-images.githubusercontent.com/12170745/31783293-c540ec1c-b4cb-11e7-8e2d-0a10c287ae11.png)

Shouldn't the left-hand column read like this?
```
99 ("top 1%")
95-98
90-94
80-89
70-79
60-69
50-59
40-49
30-39
20-29
10-19
0-9
```

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @hdoupe @GoFroggyRun

**Comment by MattHJensen** [comment]
I don't think it is incorrect per se. It looks like they are doing  (x-y] and therefore starting with 1 instead of 0.

**Comment by martinholmer** [comment]
@MattHJensen said:
> I don't think it is incorrect per se. It looks like they are doing (x-y] and therefore starting with 1 instead of 0.

I guess numbers in the left-hand column supposed to be **quantiles** that are the boundaries of the percentiles.  Does that make sense?

**Comment by MattHJensen** [comment]
> I guess numbers in the left-hand column supposed to be quantiles that are the boundaries of the percentiles. Does that make sense?

Yep, that's what I'm thinking. The left-hand column is a an exclusive boundary, and the right-hand column is an inclusive boundary.

**Comment by MattHJensen** [comment]
It lines up better with the way people talk about the top 1% and bottom 99%, I suppose.

**Comment by martinholmer** [comment]
Issue #1374 has been resolved by merge of #1602.
